.pre-commit-config.yaml

```diff
@@ -0,0 +1,37 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v3.2.0
+  hooks:
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: check-yaml
+    - id: check-json
+    - id: check-toml
+    - id: pretty-format-json
+    - id: check-added-large-files
+    - id: check-case-conflict
+    - id: check-executables-have-shebangs
+    - id: check-merge-conflict
+    - id: detect-private-key
+- repo: https://github.com/hackaugusto/pre-commit-cargo
+  rev: v1.0.0
+  hooks:
+    # Allows cargo fmt to modify the source code prior to the commit
+    - id: cargo
+      name: Cargo fmt
+      args: ["+nightly", "fmt", "--all"]
+      stages: [commit]
+    # Requires code to be properly formatted prior to pushing upstream
+    - id: cargo
+      name: Cargo fmt --check
+      args: ["+nightly", "fmt", "--all", "--check"]
```

Comment:
we are moving `rustfmt` to `nightly`. it's 2023, so we shouldn't have toolchain issues by doing that; but, historically, rustfmt was often unavailable as nightly and this frequently broke the CI.

while we shouldn't expect such instability to happen again, is it worthy the risk? `rustfmt` probably won't need nightly items anyway.

Reply:
I don't have strong preferences here, I only did this because clippy is also using nightly. Should I move both to stable?